### PR TITLE
e2e: Wait for the registry Pod to sync before deleting the test CatalogSource

### PR DESCRIPTION
**Pull request by timflannagan**
Signed-off-by: timflannagan <timflannagan@gmail.com>

**Description of the change:**
Update this metrics e2e test to first wait until the testing CatalogSource's underlying registry Pod is reporting a successful/synced state before attempting to delete it. This helps cut down on the potential for creating this CatalogSource, then immediately deleting it, which can strand the underlying registry Pod is a terminating state for a couple of minutes (and therefore exceeding the wait time for this check). This was noticed in a downstream CI environment, and was consistently reproducible:

<details>

```console
$ kubens openshift-operators
$ k logs metrics-catsrc-valid-chwd8  -c configmap-registry-server
Error from server (BadRequest): container "configmap-registry-server" in pod "metrics-catsrc-valid-chwd8" is waiting to start: ContainerCreating
$ k get metrics-catsrc-valid-chwd8 -o yaml
...
  containerStatuses:
  - image: registry.build01.ci.openshift.org/ci-ln-cpx8s2b/stable@sha256:9b03603d7c64fdd1c3aaddde3c6ce1e7110c9e6838e071acfc8414fcf42fb373
    imageID: ""
    lastState: {}
    name: configmap-registry-server
    ready: false
    restartCount: 0
    started: false
    state:
      waiting:
        reason: ContainerCreating
$ k get events --sort-by='.metadata.creationTimestamp'
LAST SEEN   TYPE      REASON        OBJECT                           MESSAGE
9m7s        Normal    Scheduled     pod/metrics-catsrc-valid-9qs2t   Successfully assigned openshift-operators/metrics-catsrc-valid-9qs2t to ci-ln-cpx8s2b-72292-f52jz-worker-a-788cb
8m3s        Warning   FailedMount   pod/metrics-catsrc-valid-9qs2t   MountVolume.SetUp failed for volume "kube-api-access-nnvcz" : failed to fetch token: serviceaccounts "metrics-catsrc-valid-configmap-server" not found
7m4s        Warning   FailedMount   pod/metrics-catsrc-valid-9qs2t   Unable to attach or mount volumes: unmounted volumes=[kube-api-access-nnvcz], unattached volumes=[kube-api-access-nnvcz]: timed out waiting for the condition
4m36s       Normal    Scheduled     pod/metrics-catsrc-valid-chwd8   Successfully assigned openshift-operators/metrics-catsrc-valid-chwd8 to ci-ln-cpx8s2b-72292-f52jz-worker-a-788cb
3m32s       Warning   FailedMount   pod/metrics-catsrc-valid-chwd8   MountVolume.SetUp failed for volume "kube-api-access-jpxds" : failed to fetch token: serviceaccounts "metrics-catsrc-valid-configmap-server" not found
2m33s       Warning   FailedMount   pod/metrics-catsrc-valid-chwd8   Unable to attach or mount volumes: unmounted volumes=[kube-api-access-jpxds], unattached volumes=[kube-api-access-jpxds]: timed out waiting for the condition
```

</details>

Because the downstream cluster we're running on is fairly bloated compared to KinD clusters, we risk ginkgo ordering this spec such that we're creating this CS resource, then immediately deleting it before the catalog-operator has time to generate all the requisite resources (e.g. RBAC/ServiceAccount/etc.), and so the test case times out waiting for the underlying Pod to be fully deleted:

<details>

```log
[BeforeEach] Metrics are generated for OLM managed resources
  /go/src/github.com/openshift/operator-framework-olm/staging/operator-lifecycle-manager/test/e2e/metrics_e2e_test.go:39
[BeforeEach] when A valid CatalogSource object is created
  /go/src/github.com/openshift/operator-framework-olm/staging/operator-lifecycle-manager/test/e2e/metrics_e2e_test.go:328
Creating catalog source metrics-catsrc-valid in namespace openshift-operators...
Catalog source metrics-catsrc-valid created
[BeforeEach] when The CatalogSource object is deleted
  /go/src/github.com/openshift/operator-framework-olm/staging/operator-lifecycle-manager/test/e2e/metrics_e2e_test.go:374
Deleting config map metrics-catsrc-valid-configmap...
Deleting catalog source metrics-catsrc-valid...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
waiting for the catalog source metrics-catsrc-valid-9ms5w pod to be deleted...
[AfterEach] when A valid CatalogSource object is created
  /go/src/github.com/openshift/operator-framework-olm/staging/operator-lifecycle-manager/test/e2e/metrics_e2e_test.go:354
```

</details>

**Motivation for the change:**
Burn down CI flakes/failures to improve CI signal.

**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage
- [ ] Sufficient end-to-end test coverage
- [ ] Docs updated or added to `/doc`
- [ ] Commit messages sensible and descriptive
- [ ] Tests marked as `[FLAKE]` are truly flaky
- [ ] Tests that remove the `[FLAKE]` tag are no longer flaky

### CODE REVIEW

File: test/e2e/metrics_e2e_test.go [hunk 1]
```diff
@@ -1,3 +1,4 @@
+//go:build !bare
```
Comment (akihikokuroda) [inline test/e2e/metrics_e2e_test.go:1]: I'm not familiar with this.  What is this for?

Comment (timflannagan) [reply test/e2e/metrics_e2e_test.go:1]: oh weird - I'm not sure what happened there? it looks like my IDE automatically added this so I'll have to dive more into this later.

Comment (timflannagan) [reply test/e2e/metrics_e2e_test.go:1]: Ah, it looks like this build tag was already present but a recent Go minor version changed the structure, so the diff is misleading.

**Pull request merged**